test/numa/run.sh

```diff
@@ -419,6 +432,17 @@ create() { # script API
     done
 }
 
+reset() { # script API
+    # Usage: reset counters
+    #
+    # Resets counters
+    if [ "$1" == "counters" ]; then
+        kind_count[pod]=0
+    else
+        error "invalid reset \"$1\""
+    fi
+}
+
```

Comment:
Could we add something like this below for `run.sh:create()` ? Otherwise it still forces folks to run it from a particular directory. As an added benefit this would allow policies, test topologies, and tests to override the default templates provided in $SCRIPT_DIR as/if necessary.

```
diff --git a/test/numa/run.sh b/test/numa/run.sh
index 7689a628..5dca9c65 100755
--- a/test/numa/run.sh
+++ b/test/numa/run.sh
@@ -388,12 +388,38 @@ delete() { # script API
     }
 }
 
+resolve-template() {
+    local name="$1" r="" d t
+    shift
+    for d in $*; do
+        if [ -z "$d" ] || ! [ -d "$d" ]; then
+            continue
+        fi
+        t=$d/$name.yaml.in
+        if ! [ -e $t ]; then
+            continue
+        fi
+        if [ -z "$r" ]; then
+            r="$t"
+        else
+            echo "WARNING: template file $r shadows $t"
+        fi
+    done
+    if [ -n "$r" ]; then
+        echo "$r"
+        return 0
+    fi
+    return 1
+}
+
 create() { # script API
     # Usage: [VAR=VALUE][n=COUNT] create TEMPLATE_NAME
     #
     # Create n instances from TEMPLATE_NAME.yaml.in, copy each of them
     # from host to vm, kubectl create -f them, and wait for them
-    # becoming Ready.
+    # becoming Ready. Templates are searched in $TEST_DIR, $TOPOLOGY_DIR,
+    # $POLICY_DIR, and $SCRIPT_DIR in this order of preference. The first
+    # template found is used.
     #
     # Parameters:
     #   TEMPLATE_NAME: the name of the template without extension (.yaml.in)
@@ -402,21 +428,23 @@ create() { # script API
     #   wait: condition to be waited for (see kubectl wait --for=condition=).
     #         If empty (""), skip waiting. The default is wait="Ready".
     #   wait_t: wait timeout. The default is wait_t=60s.
-    template_name=$1.yaml.in
+
+    template_file=$(resolve-template $1 $TEST_DIR $TOPOLOGY_DIR $POLICY_DIR $SCRIPT_DIR)
+
     local template_kind
-    template_kind=$(awk '/kind/{print tolower($2)}' < "$template_name")
+    template_kind=$(awk '/kind/{print tolower($2)}' < "$template_file")
     local wait=${wait-Ready}
     local wait_t=${wait_t-60s}
     if [ -z "$n" ]; then
         local n=1
     fi
-    if [ ! -f "$template_name" ]; then
-        error "error creating \"$1\": missing template ${template_name}"
+    if [ ! -f "$template_file" ]; then
+        error "error creating \"$1\": missing template ${template_file}"
     fi
     for _ in $(seq 1 $n); do
         kind_count[$template_kind]=$(( ${kind_count[$template_kind]} + 1 ))
         local NAME="${template_kind}$(( ${kind_count[$template_kind]} - 1 ))" # the first pod is pod0
-        eval "echo -e \"$(<"${template_name}")\"" > "$NAME.yaml"
+        eval "echo -e \"$(<"${template_file}")\"" > "$NAME.yaml"
         host-command "scp $NAME.yaml $VM_SSH_USER@$VM_IP:" || {
             command-error "copying $NAME.yaml to VM failed"
         }

```